koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -536,41 +451,192 @@ def check_whitelist(self, name, target_info):
-                  compose_ids=None, skip_build=False, triggered_after_koji_task=None):
+    def handle_build_response(self, build_response, arch=None, auto_msg=False):
+        build_id = build_response.get_build_name()
+        self.logger.debug("OSBS build id: %r", build_id)
 
-        self.logger.debug("Spawning jobs for arches: %r" % (arches))
+        # When builds are cancelled the builder plugin process gets SIGINT and SIGKILL
+        # If osbs has started a build it should get cancelled
+        def sigint_handler(*args, **kwargs):
+            if not build_id:
+                return
 
-        results = []
+            self.logger.warn("Cannot read logs, cancelling build %s", build_id)
+            self.osbs().cancel_build(build_id)
 
-        kwargs = dict(
-            src=src,
-            target_info=target_info,
-            scratch=scratch,
-            isolated=isolated,
-            yum_repourls=yum_repourls,
-            branch=branch,
-            push_url=push_url,
-            arches=arches,
-            koji_parent_build=koji_parent_build,
-            release=release,
-            flatpak=flatpak,
-            signing_intent=signing_intent,
-            compose_ids=compose_ids,
-            skip_build=skip_build,
-            triggered_after_koji_task=triggered_after_koji_task
-        )
+        signal.signal(signal.SIGINT, sigint_handler)
 
-        results = []
-        semi_results = self.createContainer(**kwargs)
-        if semi_results is not None:
-            results = [semi_results]
+        self.logger.debug("Waiting for osbs build_id: %s to be scheduled.",
+                          build_id)
+        # we need to wait for kubelet to schedule the build, otherwise it's 500
+        self.osbs().wait_for_build_to_get_scheduled(build_id)
+        self.logger.debug("Build was scheduled")
+
+        osbs_logs_dir = self.resultdir()
+        koji.ensuredir(osbs_logs_dir)
+        pid = os.fork()
+        if pid:
+            try:
+                self._incremental_upload_logs(pid)
+            except koji.ActionNotAllowed:
+                pass
+        else:
+            self._osbs = None
+
+            try:
+                self._write_incremental_logs(build_id, osbs_logs_dir)
+            except Exception as error:
+                self.logger.info("Error while saving incremental logs: %s", error)
+                os._exit(1)
+            os._exit(0)
+
+        response = self.osbs().wait_for_build_to_finish(build_id)
+
+        self.logger.debug("OSBS build finished with status: %s. Build "
+                          "response: %s.", response.status,
+                          response.json)
+
+        self.logger.info("Response status: %r", response.is_succeeded())
+
+        if response.is_cancelled():
+            self.session.cancelTask(self.id)
+
+            if auto_msg:
+                cancel_msg = 'Image build was cancelled by OSBS, maybe by automated rebuild.'
```

Comment:
Thinking about this some more... do automated rebuilds ever cancel other builds? I don't see any code that would do that.
I wonder if this "maybe by automated rebuild" suggestion was just wrong even before this PR.

Reply:
I'll remove it, I think it was from the past, when we wanted autorebuilds cancel regular builds